### PR TITLE
Add more data to PaymentFailed event

### LINKED ISSUE #143: Missing payment reference with failed payment event
Currently the payment failure event only emits an error string without reference to the original payment. To resolve which payment has failed a client will also need a reference to the payment, be it the original payment object or payment hash.

### PR DESCRIPTION
**Pull request by roeierez**
This PR adds the following data to the payment failed event:
1. Node ID
2. Optional parsed bolt11 invoice.

fixed #143

### CODE REVIEW

File: libs/sdk-bindings/src/breez_sdk.udl [hunk 1]
```diff
@@ -211,13 +211,19 @@ dictionary InvoicePaidDetails {
     string bolt11;
 };
 
+dictionary PaymentFailedData {
+    string error;
+    string node_id;
+    LNInvoice? bolt11;
```
Comment (dangeross) [inline libs/sdk-bindings/src/breez_sdk.udl:217]: bolt11 is usually a string with the encoded payment request. Would in be clearer to rename the lnInvoice param to invoice?

File: libs/sdk-react-native/ios/BreezSDKMapper.swift [hunk 2]
```diff
@@ -172,6 +172,14 @@ class BreezSDKMapper {
             "bolt11": invoicePaidDetails.bolt11
         ]
     }
+
+    static func dictionaryOf(paymentFailedData: PaymentFailedData) -> [String: Any?] {
+        return [
+            "error": paymentFailedData.error,
+            "bolt11": dictionaryOf(lnInvoice: paymentFailedData.bolt11),
```
Comment (dangeross) [inline libs/sdk-react-native/ios/BreezSDKMapper.swift:179]: If bolt11 is optional then `dictionaryOf(lnInvoice)` needs to handle the nil case.

File: libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKMapper.kt [hunk 3]
```diff
@@ -190,6 +190,14 @@ fun readableMapOf(invoicePaidDetails: InvoicePaidDetails): ReadableMap {
     )
 }
 
+fun readableMapOf(paymentFailedData: PaymentFailedData): ReadableMap {
+    return readableMapOf(
+            "error" to paymentFailedData.error,
+            "bolt11" to readableMapOf(paymentFailedData.bolt11)
```
Comment (dangeross) [inline libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKMapper.kt:196]: If bolt11 is optional then `readableMapOf(lnInvoice: LnInvoice)` needs to handle the null case.

**Review by ok300 — APPROVED** [review]
The Rust side LGTM.

File: libs/sdk-bindings/src/breez_sdk.udl (hunk 1 above)
Comment (roeierez) [reply libs/sdk-bindings/src/breez_sdk.udl]: Agree. Done

File: libs/sdk-react-native/ios/BreezSDKMapper.swift (hunk 2 above)
Comment (roeierez) [reply libs/sdk-react-native/ios/BreezSDKMapper.swift]: Right, missed that. Fixed.

File: libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKMapper.kt (hunk 3 above)
Comment (roeierez) [reply libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKMapper.kt]: Right, missed that. Fixed.

**Review by dangeross — APPROVED** [review]
LGTM!

**Pull request merged**